synapse/federation/sender/per_destination_queue.py

```diff
@@ -347,32 +343,33 @@ def _pop_pending_edus(self, limit):
 
     @defer.inlineCallbacks
     def _get_new_device_messages(self):
-        last_device_stream_id = self._last_device_stream_id
-        to_device_stream_id = self._store.get_to_device_stream_token()
-        contents, stream_id = yield self._store.get_new_device_msgs_for_remote(
-            self._destination, last_device_stream_id, to_device_stream_id
+        last_device_list = self._last_device_list_stream_id
+        now_stream_id, results = yield self._store.get_devices_by_remote(
+            self._destination, last_device_list
         )
         edus = [
             Edu(
                 origin=self._server_name,
                 destination=self._destination,
-                edu_type="m.direct_to_device",
+                edu_type="m.device_list_update",
                 content=content,
             )
-            for content in contents
+            for content in results
         ]
 
-        last_device_list = self._last_device_list_stream_id
-        now_stream_id, results = yield self._store.get_devices_by_remote(
-            self._destination, last_device_list
+        last_device_stream_id = self._last_device_stream_id
+        to_device_stream_id = self._store.get_to_device_stream_token()
+        contents, stream_id = yield self._store.get_new_device_msgs_for_remote(
+            self._destination, last_device_stream_id, to_device_stream_id, 98 - len(edus)
```

Comment:
Here we can set a limit to fill the `edus` list. I chose 98 instead of 100 as `presence` and `rr_edus` could both take one slot after that.

Reply:
I think, rather than hardcode 98 here, it would be clearer to make `_get_new_device_messages` take a `limit` parameter.

Reply:
Also: the default value for the `limit` parameter on `get_new_device_msgs_for_remote` is now redundant. I suggest removing the default value for clarity.